src/reducer.js

```diff
@@ -172,7 +172,10 @@ export const timestampsReducer = (state = {}, { type, path }) => {
 const createDataReducer = (actionKey = 'data') => (state = {}, action) => {
   switch (action.type) {
     case SET:
-      return set(getDotStrPath(action.path), action[actionKey], state)
```

Comment:
Set overwrites the existing data at the path you are writing to. For example, if you are writing a bunch of new users from /public/users, any previous users in firebase.data.public.users will be overwritten rather than merged with.